### PR TITLE
Updating autorest.md with TLS 1.3 changes

### PR DESCRIPTION
**Pull request by jha1bhavna**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

**Review by jsquire — DISMISSED** [review]
@jha1bhavna:

Your account is not configured correctly and, as a result, this PR cannot move forward. 
 Please review the Azure SDK onboarding documentation and ensure that your GitHub organization membership and permissions are correct.
- [Azure SDK onboarding](https://eng.ms/docs/products/azure-developer-experience/onboard/requirements) _(Microsoft internal)_
- [Azure SDK onboarding assistance](https://teams.microsoft.com/l/channel/19%3ade3fce22c2994be18cac50502c55f717%40thread.skype/Azure%2520SDK%2520Onboarding?groupId=3e17dcb0-4257-4a30-b843-77f47f1d4121&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47) _(Microsoft internal)_

Please also be sure to add yourself to [CODEOWNERS](https://github.com/Azure/azure-sdk-for-net/blob/main/.github/CODEOWNERS) for this library, if you will be maintaining it going forward.

File: sdk/sqlmanagement/Azure.ResourceManager.Sql/src/Custom/Models/SqlServerPatch.cs
Comment (ArcturusZhang) [inline sdk/sqlmanagement/Azure.ResourceManager.Sql/src/Custom/Models/SqlServerPatch.cs:1]: Please format this file

File: sdk/sqlmanagement/Azure.ResourceManager.Sql/src/Custom/SqlServerData.cs
Comment (ArcturusZhang) [inline sdk/sqlmanagement/Azure.ResourceManager.Sql/src/Custom/SqlServerData.cs:1]: Please format this file

File: sdk/sqlmanagement/Azure.ResourceManager.Sql/src/Generated/Models/MinimalTlsVersion.cs
Comment (ArcturusZhang) [inline sdk/sqlmanagement/Azure.ResourceManager.Sql/src/Generated/Models/MinimalTlsVersion.cs:40]: we need to introduce customized code to change the name of these properties, the generator cannot handle them properly.

Comment (ArcturusZhang) [reply sdk/sqlmanagement/Azure.ResourceManager.Sql/src/Generated/Models/MinimalTlsVersion.cs:40]: Please introduce a partial class of this type, and put the following code (copy the generated code here and modify the part we need):
```csharp
public partial struct MinimalTlsVersion
{
	/// <summary> 1.0. </summary>
	[CodeGenMember("One0")]
        public static MinimalTlsVersion Tls1_0 { get; } = new MinimalTlsVersion(Tls1_0Value);
        /// <summary> 1.1. </summary>
	[CodeGenMember("One1")]
        public static MinimalTlsVersion Tls1_1 { get; } = new MinimalTlsVersion(Tls1_1Value);
        /// <summary> 1.2. </summary>
	[CodeGenMember("One2")]
        public static MinimalTlsVersion Tls1_2 { get; } = new MinimalTlsVersion(Tls1_2Value);
        /// <summary> 1.3. </summary>
	[CodeGenMember("One3")]
        public static MinimalTlsVersion Tls1_3 { get; } = new MinimalTlsVersion(Tls1_3Value);
}
```
then regenerate the whole library.
Do not worry if there is compilation errors when you wrote this, it should compile after you added this class and regenerate.

**Review by ArcturusZhang — COMMENTED** [review]
Could you run the script
`<root>/eng/Export-API.ps1 sqlmanagement` to generate the api file as well so that we could easily see the api changes for review?

Please note that every time you changed something in the code, you should run this script, otherwise the pipeline will report this.

File: sdk/sqlmanagement/Azure.ResourceManager.Sql/src/Custom/MinimalTlsVersion.cs
Comment (ArcturusZhang) [inline sdk/sqlmanagement/Azure.ResourceManager.Sql/src/Custom/MinimalTlsVersion.cs:48]: we only need to keep these, others in this partial class could be removed.

File: sdk/sqlmanagement/Azure.ResourceManager.Sql/src/Custom/SqlServerData.cs
Comment (jha1bhavna) [reply sdk/sqlmanagement/Azure.ResourceManager.Sql/src/Custom/SqlServerData.cs:1]: what formatting does these files need? Indentation is correct only.

File: sdk/sqlmanagement/Azure.ResourceManager.Sql/api/Azure.ResourceManager.Sql.netstandard2.0.cs
Comment (ArcturusZhang) [inline sdk/sqlmanagement/Azure.ResourceManager.Sql/api/Azure.ResourceManager.Sql.netstandard2.0.cs:720]: Could we rename this type to `SqlJobPrivateEndpointData`?

File: sdk/sqlmanagement/Azure.ResourceManager.Sql/api/Azure.ResourceManager.Sql.netstandard2.0.cs
Comment (ArcturusZhang) [inline sdk/sqlmanagement/Azure.ResourceManager.Sql/api/Azure.ResourceManager.Sql.netstandard2.0.cs:2259]: this name is quite generic, could we rename it to be more specific? We could have `SqlMinimalTlsVersion` or you could add other things as the context.

Comment (ArcturusZhang) [reply sdk/sqlmanagement/Azure.ResourceManager.Sql/api/Azure.ResourceManager.Sql.netstandard2.0.cs:2259]: Please note that since we have a customization code for this model already, after you change the name of this by using `rename-mapping`, we should also update the type names in your customization code to align the new name.

File: sdk/sqlmanagement/Azure.ResourceManager.Sql/api/Azure.ResourceManager.Sql.netstandard2.0.cs
Comment (ArcturusZhang) [inline sdk/sqlmanagement/Azure.ResourceManager.Sql/api/Azure.ResourceManager.Sql.netstandard2.0.cs:6191]: We should add a context as prefix of the type name of this type as well. We could simply use `Sql`

File: sdk/sqlmanagement/Azure.ResourceManager.Sql/api/Azure.ResourceManager.Sql.netstandard2.0.cs
Comment (jha1bhavna) [reply sdk/sqlmanagement/Azure.ResourceManager.Sql/api/Azure.ResourceManager.Sql.netstandard2.0.cs:720]: I don't have any changes related to Private Endpoint, so I'll prefer to not change those. Let me know your thoughts on this?

File: sdk/sqlmanagement/Azure.ResourceManager.Sql/api/Azure.ResourceManager.Sql.netstandard2.0.cs
Comment (jha1bhavna) [reply sdk/sqlmanagement/Azure.ResourceManager.Sql/api/Azure.ResourceManager.Sql.netstandard2.0.cs:2259]: Earlier also it was MinimalTlsVersion so in that sense minTlsVersion should be ok and more similar to the original name I feel.

File: sdk/sqlmanagement/Azure.ResourceManager.Sql/api/Azure.ResourceManager.Sql.netstandard2.0.cs
Comment (jha1bhavna) [reply sdk/sqlmanagement/Azure.ResourceManager.Sql/api/Azure.ResourceManager.Sql.netstandard2.0.cs:6191]: I don't have any changes related to BackupStorage nor do I have knowledge of that, this has got autogenerated. I'll prefer to not touch that area.

File: sdk/sqlmanagement/Azure.ResourceManager.Sql/api/Azure.ResourceManager.Sql.netstandard2.0.cs
Comment (ArcturusZhang) [reply sdk/sqlmanagement/Azure.ResourceManager.Sql/api/Azure.ResourceManager.Sql.netstandard2.0.cs:720]: This shows up here meaning that your PR (this PR) introduces these new types.
Once we release, this new type will release out and we are not able to change its name any more, therefore we have to change it here.

File: sdk/sqlmanagement/Azure.ResourceManager.Sql/api/Azure.ResourceManager.Sql.netstandard2.0.cs
Comment (ArcturusZhang) [reply sdk/sqlmanagement/Azure.ResourceManager.Sql/api/Azure.ResourceManager.Sql.netstandard2.0.cs:2259]: We would like type names to be more consistent between libraries as well, usually this is called XXXMinimalTlsVersion which is the reason I suggest `SqlMinimalTlsVersion`. The name `MinTlsVersion` is an unfortunate workaround.

File: sdk/sqlmanagement/Azure.ResourceManager.Sql/src/Custom/MinimalTlsVersion.cs
Comment (ArcturusZhang) [inline sdk/sqlmanagement/Azure.ResourceManager.Sql/src/Custom/MinimalTlsVersion.cs:1]: This file does not have a proper indentation, could we format this file?

File: sdk/sqlmanagement/Azure.ResourceManager.Sql/src/Custom/MinimalTlsVersion.cs
Comment (ArcturusZhang) [inline sdk/sqlmanagement/Azure.ResourceManager.Sql/src/Custom/MinimalTlsVersion.cs:32]: and we could remove all of these. The generator will generate them.

File: sdk/sqlmanagement/Azure.ResourceManager.Sql/src/Custom/Models/SqlServerPatch.cs
Comment (ArcturusZhang) [inline sdk/sqlmanagement/Azure.ResourceManager.Sql/src/Custom/Models/SqlServerPatch.cs:1]: this file needs a format as well

File: sdk/sqlmanagement/Azure.ResourceManager.Sql/src/Custom/SqlServerData.cs
Comment (ArcturusZhang) [inline sdk/sqlmanagement/Azure.ResourceManager.Sql/src/Custom/SqlServerData.cs:1]: this too needs to format

File: sdk/sqlmanagement/Azure.ResourceManager.Sql/src/Custom/MinimalTlsVersion.cs
Comment (jha1bhavna) [reply sdk/sqlmanagement/Azure.ResourceManager.Sql/src/Custom/MinimalTlsVersion.cs:32]: It is giving this error if I remove them: 
Build FAILED.

D:\src\NET_sdk\azure-sdk-for-net\sdk\sqlmanagement\Azure.ResourceManager.Sql\src\Custom\MinimalTlsVersion.cs(21,88): er
ror CS0103: The name 'Tls_None' does not exist in the current context [D:\src\NET_sdk\azure-sdk-for-net\sdk\sqlmanageme
nt\Azure.ResourceManager.Sql\src\Azure.ResourceManager.Sql.csproj::TargetFramework=netstandard2.0]
D:\src\NET_sdk\azure-sdk-for-net\sdk\sqlmanagement\Azure.ResourceManager.Sql\src\Custom\MinimalTlsVersion.cs(21,67): er
ror CS1729: 'SqlMinimalTlsVersion' does not contain a constructor that takes 1 arguments [D:\src\NET_sdk\azure-sdk-for-
net\sdk\sqlmanagement\Azure.ResourceManager.Sql\src\Azure.ResourceManager.Sql.csproj::TargetFramework=netstandard2.0]
D:\src\NET_sdk\azure-sdk-for-net\sdk\sqlmanagement\Azure.ResourceManager.Sql\src\Custom\MinimalTlsVersion.cs(24,87): er
ror CS0103: The name 'Tls1_0Value' does not exist in the current context [D:\src\NET_sdk\azure-sdk-for-net\sdk\sqlmanag
ement\Azure.ResourceManager.Sql\src\Azure.ResourceManager.Sql.csproj::TargetFramework=netstandard2.0]
D:\src\NET_sdk\azure-sdk-for-net\sdk\sqlmanagement\Azure.ResourceManager.Sql\src\Custom\MinimalTlsVersion.cs(24,66): er
ror CS1729: 'SqlMinimalTlsVersion' does not contain a constructor that takes 1 arguments [D:\src\NET_sdk\azure-sdk-for-
net\sdk\sqlmanagement\Azure.ResourceManager.Sql\src\Azure.ResourceManager.Sql.csproj::TargetFramework=netstandard2.0]
D:\src\NET_sdk\azure-sdk-for-net\sdk\sqlmanagement\Azure.ResourceManager.Sql\src\Custom\MinimalTlsVersion.cs(27,87): er
ror CS0103: The name 'Tls1_1Value' does not exist in the current context [D:\src\NET_sdk\azure-sdk-for-net\sdk\sqlmanag
ement\Azure.ResourceManager.Sql\src\Azure.ResourceManager.Sql.csproj::TargetFramework=netstandard2.0]
D:\src\NET_sdk\azure-sdk-for-net\sdk\sqlmanagement\Azure.ResourceManager.Sql\src\Custom\MinimalTlsVersion.cs(27,66): er
ror CS1729: 'SqlMinimalTlsVersion' does not contain a constructor that takes 1 arguments [D:\src\NET_sdk\azure-sdk-for-
net\sdk\sqlmanagement\Azure.ResourceManager.Sql\src\Azure.ResourceManager.Sql.csproj::TargetFramework=netstandard2.0]
D:\src\NET_sdk\azure-sdk-for-net\sdk\sqlmanagement\Azure.ResourceManager.Sql\src\Custom\MinimalTlsVersion.cs(30,87): er
ror CS0103: The name 'Tls1_2Value' does not exist in the current context [D:\src\NET_sdk\azure-sdk-for-net\sdk\sqlmanag
ement\Azure.ResourceManager.Sql\src\Azure.ResourceManager.Sql.csproj::TargetFramework=netstandard2.0]
D:\src\NET_sdk\azure-sdk-for-net\sdk\sqlmanagement\Azure.ResourceManager.Sql\src\Custom\MinimalTlsVersion.cs(30,66): er
ror CS1729: 'SqlMinimalTlsVersion' does not contain a constructor that takes 1 arguments [D:\src\NET_sdk\azure-sdk-for-
net\sdk\sqlmanagement\Azure.ResourceManager.Sql\src\Azure.ResourceManager.Sql.csproj::TargetFramework=netstandard2.0]
D:\src\NET_sdk\azure-sdk-for-net\sdk\sqlmanagement\Azure.ResourceManager.Sql\src\Custom\MinimalTlsVersion.cs(33,87): er
ror CS0103: The name 'Tls1_3Value' does not exist in the current context [D:\src\NET_sdk\azure-sdk-for-net\sdk\sqlmanag
ement\Azure.ResourceManager.Sql\src\Azure.ResourceManager.Sql.csproj::TargetFramework=netstandard2.0]
D:\src\NET_sdk\azure-sdk-for-net\sdk\sqlmanagement\Azure.ResourceManager.Sql\src\Custom\MinimalTlsVersion.cs(33,66): er
ror CS1729: 'SqlMinimalTlsVersion' does not contain a constructor that takes 1 arguments [D:\src\NET_sdk\azure-sdk-for-
net\sdk\sqlmanagement\Azure.ResourceManager.Sql\src\Azure.ResourceManager.Sql.csproj::TargetFramework=netstandard2.0]
    0 Warning(s)
    10 Error(s)

File: sdk/sqlmanagement/Azure.ResourceManager.Sql/src/Custom/SqlServerData.cs
Comment (ArcturusZhang) [inline sdk/sqlmanagement/Azure.ResourceManager.Sql/src/Custom/SqlServerData.cs:22]: we do not need to change this. This is **property name**, not the type name

File: sdk/core/Azure.Core/src/Shared/AuthorizationChallengeParser.cs
Comment (jsquire) [inline sdk/core/Azure.Core/src/Shared/AuthorizationChallengeParser.cs:30]: Your PR should include only files that you're changing.   This does not belong here nor do many of the other changes that seem to have come in via merge commits.

Please rebase on `main` and ensure that only your intended changes are included.

**Review by ArcturusZhang — APPROVED** [review]

**Comment by ArcturusZhang** [comment]
Please update CHANGELOG.md for your changes in this PR as well

**Comment by jsquire** [comment]
@ArcturusZhang:   Do not dismiss a request for changes without explicit consent from the reviewer.  
@jha1bhavna:  Your account is not in the correct state.   You need to make your organization memberships public.

**Pull request merged**
